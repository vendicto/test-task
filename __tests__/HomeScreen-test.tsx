import { render, screen } from "@testing-library/react-native";
import HomeScreen from "@/app/index";

describe("<HomeScreen />", () => {
  test("HomeScreen renders correctly", () => {
    // Arrange & Act
    render(<HomeScreen />);

    // Assert
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
