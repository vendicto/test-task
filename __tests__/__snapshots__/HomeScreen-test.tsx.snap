// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> HomeScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 16,
    }
  }
>
  <View>
    <Text
      style={
        {
          "marginVertical": 8,
          "textAlign": "center",
        }
      }
    >
      Test app
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      href="/transactions"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="link"
      style={
        {
          "backgroundColor": "skyblue",
          "marginVertical": 8,
          "paddingHorizontal": 20,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Go to transactions
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
